prov/util/src/util_coll.c

```diff
@@ -557,77 +558,113 @@ static int util_coll_proc_reduce_item(struct util_coll_mc *coll_mc,
 	return FI_SUCCESS;
 }
 
-static int util_coll_process_work_items(struct util_coll_mc *coll_mc)
+int ofi_coll_process_pending(struct util_ep *ep)
 {
+	struct util_coll_mc *coll_mc;
+	struct slist_entry *entry;
 	struct util_coll_hdr *hdr;
 	struct util_coll_xfer_item *xfer_item;
+	struct iovec iov;
+	struct fi_msg_tagged msg = { 0 };
+	int err;
+	coll_mc = ep->av->coll_mc;
+	if (coll_mc) {
+		while (!slist_empty(&coll_mc->pending_xfer_list)) {
+			entry = slist_remove_head(&coll_mc->pending_xfer_list);
+			hdr = container_of(entry, struct util_coll_hdr, entry);
+			switch (hdr->type) {
+			case UTIL_COLL_SEND:
+				xfer_item = (struct util_coll_xfer_item *)hdr;
+				iov.iov_base = xfer_item->buf;
+				iov.iov_len = (xfer_item->count *
+					       ofi_datatype_size(
+						       xfer_item->datatype));
+				msg.msg_iov = &iov;
+				msg.iov_count = 1;
```

Comment:
The initialization of msg.msg_iov and msg.iov_count can be moved outside of all loops.